django/test/testcases.py

```diff
@@ -1345,3 +1351,26 @@ def setUpClass(cls):
     def tearDownClass(cls):
         super().tearDownClass()
         cls._lockfile.close()
+
+
+class WidgetTestCase(SimpleTestCase):
+    """
+    For testing widget rendering in Django's test suite (not a public API).
+    """
```

Comment:
Yes, the uncertainty about this situation has bothered me for a while. I wrote to [django-developers](https://groups.google.com/d/topic/django-developers/mfuo0sOF2Xo/discussion).